apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore/Logging/GoogleLogger.cs

```diff
@@ -144,6 +139,39 @@ public void Log<TState>(LogLevel logLevel, EventId eventId, TState state, Except
                 }
             }
 
+            var currentLogScope = GoogleLoggerScope.Current;
+            if (currentLogScope != null)
+            {
+                jsonStruct.Fields.Add("scope", Value.ForString(currentLogScope.ToString()));
+            }
+
+            // Create a map of format parameters of all the parent scopes,
+            // starting from the most inner scope to the top-level scope.
+            var scopeIndex = 0;
+            var scopeParamArray = new Struct();
+            while (currentLogScope != null)
+            {
+                // Determine if the state of the scope are format params
+                if (currentLogScope.State is IEnumerable<KeyValuePair<string, object>> scopeFormatParams)
+                {
+                    var scopeParams = new Struct();
+                    foreach (var pair in scopeFormatParams)
+                    {
+                        scopeParams.Fields[pair.Key] = Value.ForString(pair.Value?.ToString() ?? "");
+                    }
+
+                    scopeParamArray.Fields.Add(scopeIndex.ToString(), Value.ForStruct(scopeParams));
```

Comment:
I find this a little unexpected, having the index be the name of the field. I would rather this be a proper array (you can use `Value.ForList()` to turn an array of `Values` into a `Value`).
How it currently is, to access the elements in `scopeParams` the code needs to know if some scopes were skipped because they didn't contain scoped params, or it needs to constantly ask if a field named after a given index exists.
I understand that with an array you can't know the exact index of a scope (in the case that some are skipped) but I really don't think that that is relevant.

Reply:
My intention was indeed to create an array. I'll change this to use `Value.ForList`.